File: Assinment30&31/program7.java
//write a program which accept one number from user and off 7th & 10th bit of that number if it is on. return modified number


import java.lang.*;
import java.util.*;
//  0000  0000   0000  0000  0000  0010  0100  0000
//  0       0      0    0      0    2      4   0

class Bitwise
{
    public int OffBit(int iNo)
    {
        int iMask =0X00000240;      //576
        int iResult = 0;

        iResult = iNo ^ iMask;
        return iResult;

    }
}

class program7
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the Number");
        int iValue = sobj.nextInt();

        Bitwise bobj = new Bitwise();
        
        int iRet = bobj.OffBit(iValue);

        System.out.println("Update Number is : "+iRet);


    }
}

File: Assinment30&31/program7.cpp
//write genric program which accept N values and count frequency of any specific value

//Input :  10  20 30 10 30 40 10 40 10
// value to check frequency 10
// Output : 4

#include<iostream>
using namespace std;

template<class T>

int  Frequency(T *arr, int iSize, T iNo)
{
    int Count = 0;
    for(int i = 0; i<iSize; i++)
    {
        if(arr[i] == iNo)
        {
            Count++;
        }
    }
    return Count;
}

int main()
{
   int arr[] = {10,20,30,10,30,40,10,40,10};
   int iRet = Frequency(arr,9,10);
   cout<<"Frequency of 10 are : "<<iRet<<endl;
    return 0;
}